core/controllers/feedback.py

```diff
@@ -22,6 +22,38 @@
 from core.controllers import base
 from core.domain import feedback_services
 from core.domain import suggestion_services
+from core.domain import user_services
+
+
+def get_updated_thread_dict(entity_type, entity_id, has_suggestion):
+    """Fetches all threads by the give parameters.
+
+    Args:
+        entity_type: str. The type of entity the feedback thread is linked to.
+        entity_id: str. The id of the entity.
+        has_suggestion: bool. Whether the threads have a suggestion or not.
+
+    Returns:
+        list(dict). The updated dictionary with usernames.
+    """
+    thread_dicts = [
+        t.to_dict() for t in feedback_services.get_all_threads(
+            entity_type, entity_id, has_suggestion)]
+
+    original_author_ids = [
+        t['original_author_id'] for t in thread_dicts]
+    last_nonempty_message_author_ids = [
+        t['last_nonempty_message_author_id'] for t in thread_dicts]
```

Comment:
Can any of these be `None`?

Reply:
Sorry, I missed it before but added this time.